docs/index.adoc

```diff
@@ -11401,6 +11404,31 @@ Multi-line text blocks can be used in command and option descriptions, headers a
 
 For more details, see https://www.infoq.com/articles/java-text-blocks/[this article] by Java Language Architect Brian Goetz.
 
+=== Usage of `System.exit`
+
+When an application runs as an independent process, `System.exit` returns an exit code to the caller.
+On the other hand, when an application runs as part of the caller process, `System.exit` halts the caller process and ends up an abrupt finish.
+
+In such scenario, you need to avoid `System.exit`.
+
+For example, picocli based <<TAB Autocomplete,TAB AutoComplete application>> solves this problem by having link:autocomplete.html#_maven_example[a system property] to determine whether to call `System.exit`.
+
+Another usecase is https://www.mojohaus.org/exec-maven-plugin/[`exec-maven-plugin`] with https://www.mojohaus.org/exec-maven-plugin/java-mojo.html[`exec:java` goal].
+This goal invokes the target class on the same maven process.
+The `System.exit` call finishes maven process bypassing the rest of the maven steps including its error handlings.
+
+In picocli, any exceptions thrown during the `CommandLine.execute` method are captured, printed, and converted to an exit code.
+You can check the exit code and perform an appropriate action, such as throwing an exception, instead of calling `System.exit`.
```

Comment:
This looks strange: you are correct that picocli captures any exceptions thrown during the `CommandLine.execute` method. That is the difference with the `CommandLine.parseArgs` method. If an application wants to throw an exception when a problem occurs, it should just use `parseArgs`. 

It would be strange to first capture an exception, convert to an error code, and then use that error code to throw another, less informative, exception...
The manual should not recommend that but instead should recommend using `parseArgs`.

Reply:
changed to recommend `parseArgs` and linked to the `9.7. DIY Command Execution` section